ports/stm/supervisor/internal_flash.c

```diff
@@ -101,6 +159,85 @@ uint32_t supervisor_flash_get_block_count(void) {
 }
 
 void supervisor_flash_flush(void) {
+    if (_cache_flash_addr == NO_CACHE) return;
+
+    #if defined(STM32H7)
+    __HAL_FLASH_CLEAR_FLAG(FLASH_FLAG_ALL_ERRORS_BANK1 | FLASH_FLAG_ALL_ERRORS_BANK2);
+    #else
+    __HAL_FLASH_CLEAR_FLAG(FLASH_FLAG_EOP | FLASH_FLAG_OPERR | FLASH_FLAG_WRPERR |
+                           FLASH_FLAG_PGAERR | FLASH_FLAG_PGPERR | FLASH_FLAG_PGSERR);
+    #endif
+
+    // set up for erase
+    FLASH_EraseInitTypeDef EraseInitStruct;
+    EraseInitStruct.TypeErase = TYPEERASE_SECTORS;
+    EraseInitStruct.VoltageRange = VOLTAGE_RANGE_3; // voltage range needs to be 2.7V to 3.6V
+    // get the sector information
+    uint32_t sector_size;
+    uint32_t sector_start_addr;
+    #if defined(STM32H7)
+    EraseInitStruct.Banks = get_bank(_cache_flash_addr);
+    #endif
+    EraseInitStruct.Sector = flash_get_sector_info(_cache_flash_addr, &sector_start_addr, &sector_size);
+    EraseInitStruct.NbSectors = 1;
+    if (sector_size > sizeof(_flash_cache)) {
+        __ASM volatile ("bkpt");
+        mp_printf(&mp_plat_print, "FLASH ERR: invalid sector\n");
```

Comment:
How about just leaving the resets into safe mode here and below?

Reply:
whoops missed during cleanup